src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobContainerService.java

```diff
@@ -189,6 +190,51 @@ public void stopContainer(String containerName) {
         // for some other reason.
         // The container's main process is waiting for this file to appear and then stops the main process, thus stopping and removing the container.
         executeDockerCommandWithoutAwaitingResponse(containerId, "touch", LOCALCI_WORKING_DIRECTORY + "/stop_container.txt");
+
+        // If the container is still running, force kill it.
```

Comment:
I didn't test this but won't this always run? since we don't wait for the previous command to finish

Reply:
You're right. I forgot to remove this part but it is now gone